### PR TITLE
Fixed SSA Issues + Improved build process and unit test system

### PR DESCRIPTION
**Pull request by tgiphil**

### CODE REVIEW

**Comment by tgiphil** [comment]
Added additional functional and unit tests to the build process. Unfortunately, Qemu does not always launch the VM properly and fails. When this happens, the GitHub failed action tasks need to be re-tried.

File: Source/Mosa.Utility.UnitTests/UnitTestEngine.cs [hunk 1]
```diff
@@ -487,7 +488,7 @@ private bool WaitForReady()
 				return true;
 			}
 
-			Thread.Sleep(ConnectionDelay);
+			Thread.Sleep(10);
```
Comment (charsleysa) [inline Source/Mosa.Utility.UnitTests/UnitTestEngine.cs:491]: Is this fine hard-coded?

File: Source/Mosa.Utility.UnitTests/UnitTestEngine.cs [hunk 2]
```diff
@@ -430,7 +428,7 @@ public bool ConnectToDebugEngine()
 			{
 			}
 
-			Thread.Sleep(ConnectionDelay);
+			Thread.Sleep(10);
```
Comment (charsleysa) [inline Source/Mosa.Utility.UnitTests/UnitTestEngine.cs:431]: Is this fine hard-coded?

**Review by charsleysa — CHANGES_REQUESTED** [review]

**Review by charsleysa — APPROVED** [review]
LGTM

**Pull request merged**